app/src/org/commcare/dalvik/activities/EntityMapActivity.java

```diff
@@ -1,288 +1,186 @@
+            @Override
+            public void onMapLoaded() {
+                map.animateCamera(CameraUpdateFactory.newLatLngBounds(bounds, MAP_PADDING));
+            }
+        });
+
+        map.setOnInfoWindowClickListener(this);
+
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+                || ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
+            map.setMyLocationEnabled(true);
+        }
     }
 
     @Override
     protected void onResume() {
         super.onResume();
-        mMyLocationOverlay.enableMyLocation();
-        mMyLocationOverlay.enableCompass();
-    }
 
-    protected boolean isRouteDisplayed() {
-        return false;
+        if (mMap != null && (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+                || ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED)) {
+            mMap.setMyLocationEnabled(true);
+        }
     }
 
-    private Location getLocation() {
-        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_DENIED ||
-                ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_DENIED) {
-            // TODO PLM: warn user and ask for permissions if the user has disabled them
-            return null;
-        }
+    @Override
+    protected void onPause() {
+        super.onPause();
 
-        // Get the location manager
-        mLocationManager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
-        // Define the criteria how to select the locatioin provider -> use
-        // default
-        Criteria criteria = new Criteria();
-        String provider = mLocationManager.getBestProvider(criteria, false);
-        return mLocationManager.getLastKnownLocation(provider);
+        if (mMap != null && (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+                || ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED)) {
+            mMap.setMyLocationEnabled(false);
+        }
     }
 
-    private void setLocationBounds(Location location, double[] boundHints) {
-        // Initialize the location fields
-        if (location != null) {
-            double lat = location.getLatitude();
-            double lng = location.getLongitude();
-
-            //lLat
-            boundHints[0] = lat -1;
-            //lLng
-            boundHints[1] = lng -1;
+    @Override
+    public void onInfoWindowClick(Marker marker) {
+        Intent i = new Intent(EntityMapActivity.this.getIntent());

```

Comment:
Not sure, you should try it out. You might even be able to do `Intent i = new Intent()`. If after making that change you are still able to reach the case detail view by pressing a drop-pin, then things should be good. I suspect the only way it would break is if not using `getIntent()` leaves out some extra that was put in at some earlier point.